lib/compat/wordpress-6.5/class-wp-navigation-block-renderer.php

```diff
@@ -0,0 +1,600 @@
+<?php
+/**
+ * WP_Navigation_Block_Renderer class
+ *
+ * @package gutenberg
+ * @since 6.5.0
+ */
+
+if ( class_exists( 'WP_Navigation_Block_Renderer' ) ) {
+	return;
+}
+
+/**
+ * Helper functions used to render the navigation block.
+ */
+class WP_Navigation_Block_Renderer {
+	/**
+	 * Returns whether or not this is responsive navigation.
+	 *
+	 * @param array $attributes The block attributes.
+	 * @return bool Returns whether or not this is responsive navigation.
+	 */
+	private static function is_responsive( $attributes ) {
+		/**
+		 * This is for backwards compatibility after the `isResponsive` attribute was been removed.
+		 */
+
+		$has_old_responsive_attribute = ! empty( $attributes['isResponsive'] ) && $attributes['isResponsive'];
+		return isset( $attributes['overlayMenu'] ) && 'never' !== $attributes['overlayMenu'] || $has_old_responsive_attribute;
+	}
+
+	/**
+	 * Returns whether or not a navigation has a submenu.
+	 *
+	 * @param WP_Block_List $inner_blocks The list of inner blocks.
+	 * @return bool Returns whether or not a navigation has a submenu.
+	 */
+	private static function has_submenus( $inner_blocks ) {
+		$has_submenus = false;
+		foreach ( $inner_blocks as $inner_block ) {
+			$inner_block_content = $inner_block->render();
+			$p                   = new WP_HTML_Tag_Processor( $inner_block_content );
+			if ( $p->next_tag(
+				array(
+					'name'       => 'LI',
+					'class_name' => 'has-child',
+				)
+			) ) {
+				$has_submenus = true;
+			}
+		}
+		return $has_submenus;
+	}
+
+	/**
+	 * Determine whether to load the view script.
+	 *
+	 * @param array         $attributes   The block attributes.
+	 * @param WP_Block_List $inner_blocks The list of inner blocks.
+	 * @return bool Returns whether or not to load the view script.
+	 */
+	private static function should_load_view_script( $attributes, $inner_blocks ) {
+		$has_submenus       = WP_Navigation_Block_Renderer::has_submenus( $inner_blocks );
```

Comment:
All occurrences of
```php
WP_Navigation_Block_Renderer::function_name()
```
in this class should be replaced with
```php
static::function_name()
```
.
The `static` and `self` operators in PHP are built for this purpose and it's a common practice to use them instead of the class name when calling static functions of the same class.
```suggestion
		$has_submenus       = static::has_submenus( $inner_blocks );
```

Applies elsewhere.

Reply:
thanks, done in d3c876b84c